packages/bundler-sdk-viem/src/operations.ts

```diff
@@ -915,40 +914,89 @@ export const finalizeBundle = (
     });
   });
 
-  if (unwraps.length > 0)
+  if (unwraps.length > 0) {
     steps = simulateBundlerOperations(operations.concat(unwraps), startData, {
       slippage: 0n,
     });
+    endBundlerTokenData = getLast(steps).holdings[generalAdapter1] ?? {};
+  }
 
   // Skim any token expected to be left on the bundler.
   const skims = [] as Erc20Operations["Erc20_Transfer"][];
-  {
-    const startBundlerTokenData = steps[0].holdings[generalAdapter1] ?? {};
-    const endBundlerTokenData = getLast(steps).holdings[generalAdapter1] ?? {};
-
-    skims.push(
-      ...entries(endBundlerTokenData)
-        .filter(
-          ([token, holding]) =>
-            holding != null &&
-            holding.balance - (startBundlerTokenData[token]?.balance ?? 0n) >
-              5n,
-        )
-        .map(
-          ([address]) =>
-            ({
-              type: "Erc20_Transfer",
-              address,
-              sender: generalAdapter1,
-              args: {
-                amount: maxUint256,
-                from: generalAdapter1,
-                to: receiver,
-              },
-            }) as Erc20Operations["Erc20_Transfer"],
-        ),
-    );
-  }
+  const startBundlerTokenData = steps[0].holdings[generalAdapter1] ?? {};
+
+  skims.push(
+    ...entries(endBundlerTokenData)
+      .filter(
+        ([token, holding]) =>
+          holding != null &&
+          holding.balance - (startBundlerTokenData[token]?.balance ?? 0n) > 5n,
+      )
+      .map(
+        ([address]) =>
+          ({
+            type: "Erc20_Transfer",
+            address,
+            sender: generalAdapter1,
+            args: {
+              amount: maxUint256,
+              from: generalAdapter1,
+              to: receiver,
+            },
+          }) as const,
+      ),
+  );
```

Comment:
Equivalent but you could push every addresses in a set here and at the end push all the skims based on the Set's content